src/core/MaverickV2.sol

```diff
@@ -166,6 +159,41 @@ library FastMaverickV2Pool {
             mstore(0x40, add(0x120, add(data, swapCallbackDataLength)))
         }
     }
+
+    function fastGetFromPoolState(IMaverickV2Pool pool, uint256 pos, uint256 size) internal view returns (uint256 r) {
+        assembly ("memory-safe") {
+            let ptr := mload(0x40)
+            mstore(0x00, 0x1865c57d) // selector for `getState()`
+            if iszero(staticcall(gas(), pool, 0x1c, 0x04, ptr, 0x120)) {
+                let ptr_ := mload(0x40)
+                returndatacopy(ptr_, 0x00, returndatasize())
+                revert(ptr_, returndatasize())
+            }
+            r := mload(add(pos, ptr))
+            if or(gt(0x120, returndatasize()), shr(size, r)) { revert(0x00, 0x00) }
+        }
+    }
+
+    function fastGetTick(IMaverickV2Pool pool) internal view returns (int32 r) {
+        return int32(uint32(fastGetFromPoolState(pool, 160, 32)));
+    }
+
+    function fastGetReserveAOrB(IMaverickV2Pool pool, bool tokenAIn) internal view returns (uint128 r) {
+        return uint128(fastGetFromPoolState(pool, 32 * tokenAIn.toUint(), 128));
```

Comment:
You can use a shift here instead of `32 *`. The compiler is probably smart enough to make that optimization on its own, though